add french translation for findDOMNode (#582)
reactjs/fr.react.dev: src/content/reference/react-dom/findDOMNode.md

---

## Référence {/*reference*/}

### `findDOMNode(componentInstance)` {/*finddomnode*/}

> Appeler `findDOMNode` pour trouver le nœud du navigateur pour une instance React à [composant à base de classe](/reference/react/Component) donnée.

Author: ok pas de soucis

```js
import { findDOMNode } from 'react-dom';

const domNode = findDOMNode(componentInstance);
```

[Voir les exemples ci-dessous](#usage).

#### Paramètres {/*parameters*/}

* `componentInstance` : Une instance de la sous-classe du [`Component`](/reference/react/Component). Par exemple `this` dans un composant à base de classe.

#### Valeur renvoyée {/*returns*/}

> `findDOMNode` renvoie le plus proche nœud du DOM du navigateur dans une `componentInstance` donnée. Lorsqu'un composant fait le rendu à `null`, fait le rendu à `false`, `findDOMNode` renvoie `null`. Lorsqu'un composant fait le rendu d'une string, `findDOMNode` renvoie un nœud DOM sous forme de texte contenant cette valeur.

Author: pourquoi `string` a une vf sachant que dans la communaute dev string est beaucoup plus utilisé que chaines de caractères ?

> `findDOMNode` renvoie le plus proche nœud du DOM du navigateur dans une `componentInstance` donnée. Lorsqu'un composant fait le rendu à `null`, fait le rendu à `false`, `findDOMNode` renvoie `null`. Lorsqu'un composant fait le rendu d'une string, `findDOMNode` renvoie un nœud DOM sous forme de texte contenant cette valeur.

Reviewer: Pas si ultra dominant que ça... La VF reste bien établie, et utilisée par ailleurs partout dans cette doc VF, donc on va rester cohérent.

#### Limitations {/*caveats*/}

* Un composant peut renvoyer un tableau ou un [Fragment](/reference/react/Fragment) avec plusieurs enfants. Dans ce cas `findDOMNode`, renverra le nœud du DOM correspondant au premier enfant non-vide.

> * `findDOMNode` fonctionne seulement sur les composants montés (c'est-à-dire, les composants qui ont été placés dans le DOM). Si vous essayez d'appeler `this` sur un composant qui n'a pas encore été monté (comme l'appel de `findDOMNode()` dans `render()` sur un composant qui doit être crée), une exception sera levée.

Reviewer: ```suggestion
* `findDOMNode` fonctionne seulement sur les composants montés (c'est-à-dire les composants qui ont été placés dans le DOM). Si vous essayez de l'appeler sur un composant qui n'a pas encore été monté (comme un appel de `findDOMNode()` dans `render()` sur un composant qui n'a pas encore été créé), une exception sera levée.
```

Glissement sémantique sur "this"

> * `findDOMNode` renvoie seulement le résultat au moment de votre appel. Si un composant enfant fait plutard le rendu d'un nœud différent, vous n'avez aucun moyen d'être informé de ce changement.

Reviewer: ```suggestion
* `findDOMNode` renvoie seulement le résultat au moment de votre appel. Si un composant enfant renvoie plus tard un nœud différent, vous n'avez aucun moyen d'être informé·e de ce changement.
```

Author: pourquoi `renders` est pris  ici comme "renvoie" au lieu de return qui a été choisie comme `renvoie ou valeur renvoyée`

Reviewer: Avec "render" il faut éviter une trad unique dogmatique, qui serait "faire le rendu de", car "rendre" est inenvisageable. Le souci c'est que c'est loooong, on prête donc toujours attention au sens dans le contexte.

Si la VO n'insiste pas sur la notion technique de rendu, on substituera au mieux. Parfois ce qui compte c'est l'affichage du résultat. Parfois le fait que ça rafraîchit le résultat. Parfois le fait que ça produit du JSX.

Ici j'aurais pu dire "produire", mais le point clé ici c'est la notion de racine du résultat de rendu, donc de facto la valeur renvoyée par le composant, je propose "renvoie" sur ce coup car c'est concis tout en étant juste sur la notion importante dans ce contexte.

Author: Ha ok c'est clair maintenant pour moi

* `findDOMNode` accepte une instance de composant à base de classe, il ne peut être utilisé avec les fonctions composants.

---

## Utilisation {/*usage*/}

> ### Trouver le nœud racine du DOM d'un composant à base de classe {/*finding-the-root-dom-node-of-a-class-component*/}

Reviewer: ```suggestion
### Trouver le nœud DOM racine d'un composant à base de classe {/*finding-the-root-dom-node-of-a-class-component*/}
```

> Appeler `findDOMNode` avec une instance de [composant à base de classe](/reference/react/Component)  (généralement, `this`) pour trouver le nœud du DOM qu'il a affiché.

Reviewer: ```suggestion
Appelez `findDOMNode` avec une instance de [composant à base de classe](/reference/react/Component)  (ce sera généralement `this`) pour trouver le nœud DOM qu'il a affiché.
```

  }

  render() {
    return <input defaultValue="Bonjour" />
  }
}
```

> Ici, la variable de `input` sera défini sur l'élément DOM `<input>`. Cela vous permet de faire quelque chose avec. Par exemple, en cliquant « voir l'exemple » ci-dessous, le champ de saisi est monté, [`input.select()`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/select) sélectionne tout le text dans le champ :

Reviewer: ```suggestion
Ici, la variable `input` sera définie à l'élément DOM `<input>`. Ça vous permet de le manipuler. Par exemple, en cliquant sur « voir l'exemple » ci-dessous, le champ de saisie est monté, puis [`input.select()`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/select) sélectionnera tout le texte dans le champ :
```

  return (
    <>
      <button onClick={() => setShow(true)}>

>         Voir l'exemple

Reviewer: ```suggestion
        Voir l’exemple
```

## Alternatives {/*alternatives*/}

> ### Lecture du nœud DOM à partir d'une ref {/*reading-components-own-dom-node-from-a-ref*/}

Reviewer: ```suggestion
### Lire le nœud DOM du composant à partir d'une ref {/*reading-components-own-dom-node-from-a-ref*/}
```

Les sections Utilisation utilisent toujours des objectifs exprimés à l'infinitif, pas en forme substantive.

> Le code qui utilise `findDOMNode` est fragile parce que la connection entre le nœud JSX et le code manipulant le nœud du DOM correspondant n'est pas explicite. Par exemple, essayez d'enrober ce `<input />` dans un `<div>` :

Reviewer: ```suggestion
Un code qui utilise `findDOMNode` est fragile parce que la relation entre le nœud JSX et le code manipulant le nœud DOM correspondant n'est pas explicite. Essayez par exemple d'enrober cet `<input />` dans une `<div>` :
```

  return (
    <>
      <button onClick={() => setShow(true)}>

>         Voir l'exemple

Reviewer: ```suggestion
        Voir l’exemple
```

</Sandpack>

> Le code sera interrompu maintenant, `findDOMNode(this)` trouve le nœud `<div>` du DOM, mais le code attend un nœud DOM `<input>`. Pour éviter ce genre de problèmes, utilisez [`createRef`](/reference/react/createRef) pour gérer un nœud DOM spécifique.

Reviewer: ```suggestion
Ce code plantera parce que désormais, `findDOMNode(this)` trouvera le nœud DOM `<div>` alors qu'il s'attend à un nœud DOM `<input>`. Pour éviter ce genre de problème, utilisez [`createRef`](/reference/react/createRef) pour gérer un nœud DOM spécifique.
```

> Dans cet exemple, `findDOMNode` n'est plus utilisé. Par contre, `inputRef = createRef(null)` est défini comme champ d'instance de la classe. Pour lire le nœud du DOM à partir de celui-ci, vous pouvez utiliser `this.inputRef.current`. Pour le rattacher au JSX, vous faites le rendu `<input ref={this.inputRef} />`. Cela connecte le code utilisant le nœud DOM à son JSX :

Reviewer: ```suggestion
Dans cet exemple, `findDOMNode` n'est plus utilisé. On utilise plutôt `inputRef = createRef(null)` pour définir un champ d'instance. Pour y lire le nœud DOM, vous pouvez utiliser `this.inputRef.current`. Pour le rattacher au JSX, vous mettez dans votre rendu `<input ref={this.inputRef} />`. Ça connecte le code utilisant le nœud DOM à son JSX :
```

  return (
    <>
      <button onClick={() => setShow(true)}>

>         Afficher l'exemple

Reviewer: ```suggestion
        Afficher l’exemple
```

</Sandpack>

> Dans les versions modernes de React sans les composants à base de classe, le code équivalent appelerait [`useRef`](/reference/react/useRef) à la place :

Reviewer: ```suggestion
Dans les versions modernes de React sans les composants à base de classes, le code équivalent appellerait plutôt [`useRef`](/reference/react/useRef) :
```

    input.select();
  }, []);

  return <input ref={inputRef} defaultValue="Bonjour" />
}
```

</Sandpack>

> [En savoir plus sur la manipulation du DOM avec les refs](/learn/manipulating-the-dom-with-refs).

Reviewer: ```suggestion
[Apprenez-en davantage sur la manipulation du DOM avec les refs](/learn/manipulating-the-dom-with-refs).
```

Trad normalisée

---

> ### Lecture d'un nœud DOM d'un composant enfant à partir d'un ref transmis {/*reading-a-child-components-dom-node-from-a-forwarded-ref*/}

Reviewer: ```suggestion
### Lire un nœud DOM d'un composant enfant à partir d'un ref transmis {/*reading-a-child-components-dom-node-from-a-forwarded-ref*/}
```

```js MyInput.js
export default function MyInput() {
  return <input defaultValue="Bonjour" />;
}
```

</Sandpack>

Notez que l'appel de `findDOMNode(this)` à l'intérieur de `AutoselectingInput` vous donne toujours le DOM `<input>` - même si le JSX de ce `<input>` est masqué à l'intérieur du composant `MyInput`. Cela semble pratique dans l'exemple ci-dessus, mais il conduit à un code fragile. Imaginez que vous voulez modifier `MyInput` plutard et l'enrober autour d'un `<div>`. Cela ne respectera pas le code de `AutoselectingInput` (qui attend de trouver un `<input>`).

> Pour remplacer `findDOMNode` dans cet exemple, les deux composants doivent être coordonnés :

Reviewer: ```suggestion
Pour remplacer `findDOMNode` dans cet exemple, les deux composants doivent se coordonner :
```

  return (
    <>
      <button onClick={() => setShow(true)}>

>         Afficher l'exemple

Reviewer: ```suggestion
        Afficher l’exemple
```

import { forwardRef } from 'react';

const MyInput = forwardRef(function MyInput(props, ref) {
  return <input ref={ref} defaultValue="Bonjour" />;
});

export default MyInput;
```

</Sandpack>

> Voici à quoi ressemblerait ce code avec les fonctions composants au lieu de classes :

Reviewer: ```suggestion
Voici à quoi ressemblerait ce code avec des fonctions composants au lieu de classes :
```

  return (
    <>
      <button onClick={() => setShow(true)}>

>         Afficher l'exemple

Reviewer: ```suggestion
        Afficher l’exemple
```

---

> ### Ajout d'un élement `<div>` enrobant {/*adding-a-wrapper-div-element*/}

Reviewer: ```suggestion
### Ajouter un élément `<div>` d'enrobage {/*adding-a-wrapper-div-element*/}
```

> Souvent un composant a besoin de connaître la position et la taille de ses enfants. Il est donc tentant de trouver l'enfant avec `findDOMNode(this)`, et utiliser la méthode du DOM comme [`getBoundingClientRect`](https://developer.mozilla.org/fr/docs/Web/API/Element/getBoundingClientRect) pour des mesures.

Reviewer: ```suggestion
Il arrive qu'un composant ait besoin de connaître la position et la taille de ses enfants. Il est donc tentant de trouver l'enfant avec `findDOMNode(this)`, puis d'utiliser des méthodes DOM telles que [`getBoundingClientRect`](https://developer.mozilla.org/fr/docs/Web/API/Element/getBoundingClientRect) pour faire des mesures.
```

> Il n'existe actuellement aucun équivalent direct pour ce cas d'utilisation, c'est pourquoi `findDOMNode` est déprécié mais n'a pas encore été retiré complètement de React. En attendant, vous pouvez essayez d'afficher un nœud `<div>` enrobant le contenu comme solution de contournement, et d'obtenir un ref à ce nœud. Cependant, les enveloppes supplémentaires peuvent nuire à la stylisation.

Reviewer: ```suggestion
Il n'existe actuellement aucun équivalent direct pour ce cas d'utilisation, c'est pourquoi `findDOMNode` est déprécié mais n'a pas encore été complètement retiré de React. En attendant, vous pouvez essayer d'enrober votre contenu avec un nœud `<div>` comme solution de contournement, puis d'obtenir une ref à ce nœud. Cependant, les enrobages supplémentaires peuvent nuire à l'application des styles.
```

```js
<div ref={someRef}>
  {children}
</div>
```

> Cela s'applique également à la focalisation et au défilement vers des enfants arbitraires.

Reviewer: ```suggestion
Ça s'applique également à la gestion du focus d'enfants quelconques, ou au défilement vers ces enfants.
```
